### PR TITLE
Feature/newarch

### PR DESCRIPTION
**Pull request by st235**

### CODE REVIEW

File: lib-router/src/main/java/com/github/sasd97/lib_router/BaseRouter.java [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
 
-    private Satellite satellite;
+    private final List<Satellite> satellites = new ArrayList<>();
 
     @Override
     public void pushCommand(@NonNull Command command) {
-        if (satellite == null) throw new SatelliteNotAttachedException();
-        satellite.execute(command);
+        Satellite applicableSatellite = findSatellite(command);
+        if (applicableSatellite == null) throw new SatelliteNotAttachedException();
```
Comment (Chog0) [inline lib-router/src/main/java/com/github/sasd97/lib_router/BaseRouter.java:29]: may be "SatelliteNotAddedException" would be better?

File: lib-router/src/main/java/com/github/sasd97/lib_router/commands/providers/With.java [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
+    @Nullable
+    private final Bundle bundle;
 
     private With(@NonNull Bundle bundle) {
         this.bundle = bundle;
     }
 
     private With(@NonNull ProviderCommand<Bundle> command) {
         super(command);
+        bundle = null;
```
Comment (Chog0) [inline lib-router/src/main/java/com/github/sasd97/lib_router/commands/providers/With.java:24]: why u need to make bundle null here? U can just delete final from its field

Comment (st235) [reply lib-router/src/main/java/com/github/sasd97/lib_router/commands/providers/With.java:24]: without `final` field will be modifiable 
its not true.

File: lib-router/src/main/java/com/github/sasd97/lib_router/BaseRouter.java (hunk 1 above)
Comment (st235) [reply lib-router/src/main/java/com/github/sasd97/lib_router/BaseRouter.java:29]: I would prefer `attached`

Comment (Chog0) [reply lib-router/src/main/java/com/github/sasd97/lib_router/BaseRouter.java:29]: ok

**Review by Chog0 — DISMISSED** [review]

**Comment by st235** [comment]
@Chog0, please, remove your approve.
Lets write javadoc, and replace `satellite list` to `satellite map`.

**Review by Chog0 — APPROVED** [review]

**Pull request merged**